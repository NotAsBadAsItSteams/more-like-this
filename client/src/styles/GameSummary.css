:local .element {
  background-color: #16212e;
  padding: 15px;
  width: 184px;
  border-color: #1c2938;
  border-style: solid;
  border-width: 1px;
  font-family: Tahoma, Arial, Helvetica, sans-serif;
}

:local .element:hover {
  border-color: #66c0f4;
  border-style: solid;
  border-width: 1px;
  padding: 15px;
  width: 184px;
}

:local .img {
  display: block;
  margin-left: auto;
  margin-right: auto;
}

:local .price {
  color: #acdcf5;
  font-size: 11px;
}

:local .title {
  margin-top: 5px;
  margin-bottom: 2px;
  color: #8f98a0;
  font-size: 13px;
}
