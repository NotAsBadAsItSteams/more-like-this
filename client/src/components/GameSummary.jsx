import React from 'react';
import PropTypes from 'prop-types';
import formatPrice from '../lib/humanReadable';
import styles from '../styles/GameSummary.css';

const GameSummary = ({ game: { thumbnailUrl, title, price } }) => (
  <div className={styles.element}>
    <img className={styles.img} src={thumbnailUrl} alt={title} />
    <div className={styles.title}>{title}</div>
    <div className={styles.price}>{formatPrice(price)}</div>
  </div>
);

GameSummary.propTypes = {
  game: PropTypes.shape({
    thumbnailUrl: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired,
    price: PropTypes.string.isRequired,
  }).isRequired,
};

export default GameSummary;
