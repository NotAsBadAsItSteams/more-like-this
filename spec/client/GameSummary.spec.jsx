import React from 'react';
import renderer from 'react-test-renderer';
import GameSummary from '../../client/src/components/GameSummary';

describe('GameSummary component', () => {
  test('should match snapshot', () => {
    const game = {
      price: '24.99',
      title: 'DOTA 2',
      thumbnailUrl: 'http://via.placeholder.com/184x69',
    };

    const component = renderer.create(<GameSummary game={game} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
