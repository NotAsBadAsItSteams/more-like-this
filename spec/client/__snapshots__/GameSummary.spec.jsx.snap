// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GameSummary component should match snapshot 1`] = `
<div
  className="element"
>
  <img
    alt="DOTA 2"
    className="img"
    src="http://via.placeholder.com/184x69"
  />
  <div
    className="title"
  >
    DOTA 2
  </div>
  <div
    className="price"
  >
    $24.99
  </div>
</div>
`;
